### DISCUSSION TITLE
expect.any(..) is of type Any instead of any

### DISCUSSION THREAD

**Discussion by medihack** [vitest-dev/vitest#804] [Q&A] [answered]
Vitest `expect.any(...)` (e.g. `expect.any(Number)`) returns something of type `Any`. In contrast, Jest returns `any` as type. So I always end up casting those `Any` to `any` in Vitest. Not a big problem, but I wonder if this is an explicit design decision?

**Comment by sheremet-va** [comment, marked as answer]
This is an error on our side. All asymmetric matchers should be `any`, so they could be used as a replacement in an object.
PR welcome! :)

**Reply by medihack** [reply]
I am involved in so many Open Source projects right now, unfortunately, I can't contribute to another one (even if I would like to), but I will open an issue.

**Comment by TrickyPi** [comment]
Actually, jest returns `Any` as type, see https://github.com/facebook/jest/blob/main/packages/expect/src/index.ts#L371 and https://github.com/facebook/jest/blob/main/packages/expect/src/asymmetricMatchers.ts#L336.
Did you install `@types/jest` package? Seems like it overrides `expect.any(...)` type.

**Reply by medihack** [reply]
Yes, I do use `@types/jest`. Without it, VS Code complains about missing types (using Jest ^27.5.1). So, should I close the issue? It's no big deal to do the casting manually. Just wanted to make sure that this is the expected behavior.

**Reply by TrickyPi** [reply]
yeah,`jest` need `@types/jest` in TS project. I'm a newbie to Vitest and Jest, i would like to know why you converted `Any` to `any` in Vitest.

**Reply by medihack** [reply]
If I have a larger object I want to test against I sometimes like to set it up previously to reuse it in many scenarios, e.g.

```
// somewhere else in my non-test code and imported into the test
interface Foo {
  x: number
  y: number
  z: number
}

// setup besfore test
const myFoo: Foo = {
  x: expect.any(Number), // here it would complain without "as any"
  y: 3,
  z: 7
}

// in some test
expect(bar).equals(myFoo)
```

I know it looks quite constructed, but in a larger test environment, it helps.

**Reply by TrickyPi** [reply]
Oh~, i get it, thanks for your explanation.
